src/Sarif/Baseline/DefaultBaseline/FileLocationBaselineEquals.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Baseline.DefaultBaseline
+{
+    internal class FileLocationBaselineEquals : IEqualityComparer<FileLocation>
+    {
+        public static readonly FileLocationBaselineEquals Instance = new FileLocationBaselineEquals();
+        public bool Equals(FileLocation x, FileLocation y)
+        {
+            if (!object.ReferenceEquals(x, y))
+            {
+                if (x == null || y == null)
+                {
+                    return false;
+                }
+
+                // Only compare URI (so file path/relative file path).  UriBaseId may change run over run.
```

Comment:
`uriBaseId` won't change. It will always be a token like `"SRCROOT"`. The absolute URI to which that token refers might change, but the token won't change.